src/board/v3/nap/nap_common.c

```diff
@@ -119,3 +124,55 @@ static void nap_exti_thread(void *arg)
   }
 }
 
+u8 nap_conf_rd_version_string(char version_string[])
+{
+  u8 i = 0;
+  u32 reg = 0;
+
+  do {
+    NAP->CONTROL = (i << 1);
```

Comment:
I like Gareth's suggestion of keeping the method of reading the version constant across all HDL versions so that we can gracefully warn instead of stalling the bus. Is that reasonable with this method / any changes you might want to make before committing to never changing it?